### PR TITLE
fix(engine): host's name instead of template name as alias

### PR DESCRIPTION
**Pull request by sechkem**
host's name instead of template name as alias when the alias is empty

REFS: MON-16261

### CODE REVIEW

File: common/engine_conf/parser.cc [hunk 1]
```diff
@@ -703,7 +706,8 @@ void parser::_merge(std::unique_ptr<message_helper>& msg_helper,
         } else {
           switch (f->cpp_type()) {
             case FieldDescriptor::CPPTYPE_STRING:
-              refl->SetString(msg, f, refl->GetString(*tmpl, f));
+              if (!(desc->name() == "Host" && f->name() == "alias"))
+                refl->SetString(msg, f, refl->GetString(*tmpl, f));
               break;
```
Comment (bouda1) [inline common/engine_conf/parser.cc:711]: wrong place...
the parser must be generic, it works with all types of object.

File: tests/broker-engine/hosts-with-notes-and-actions.robot [hunk 2]
```diff
@@ -105,3 +105,107 @@ EBSN3
...
+EBSN4
+    [Documentation]    New hosts with No Alias and have A Template
+    [Tags]    broker    engine    hosts    
+    Ctn Config Engine    ${1}    ${5}    ${5}    False
+    Ctn Config Broker    rrd
+    Ctn Config Broker    central
+    Ctn Config Broker    module
+    Ctn Config BBDO3    1
+    Ctn Broker Config Log    central    sql    debug
+    Ctn Config Broker Sql Output    central    unified_sql
```
Comment (bouda1) [inline tests/broker-engine/hosts-with-notes-and-actions.robot:118]: you can remove this line

File: tests/broker-engine/hosts-with-notes-and-actions.robot [hunk 3]
```diff
@@ -105,3 +105,107 @@ EBSN3
...
+    ${time_stamp2}    Evaluate    int(${time_stamp})
+    Sleep    1s
+
+    Ctn Clear Retention
+    Ctn Start Broker
+    Ctn Start engine
+
+    Connect To Database    pymysql    ${DBName}    ${DBUser}    ${DBPass}    ${DBHost}    ${DBPort}
+    FOR    ${index}    IN RANGE    10
+        ${output}    Query    SELECT alias FROM hosts WHERE name = 'host_1';
```
Comment (bouda1) [inline tests/broker-engine/hosts-with-notes-and-actions.robot:145]: the check must be in the resources table.

File: tests/broker-engine/hosts-with-notes-and-actions.robot [hunk 4]
```diff
@@ -105,3 +105,107 @@ EBSN3
...
+        Sleep    1s
+        IF    "${output}" == "(('host_1',),)"    BREAK
+    END
+    Should Be Equal As Strings    ${output}    (('host_1',),)
+
+    FOR    ${index}    IN RANGE    10
+        ${output}    Query    SELECT alias FROM hosts WHERE name = 'host_2';
```
Comment (bouda1) [inline tests/broker-engine/hosts-with-notes-and-actions.robot:152]: check in resources

File: tests/resources/Engine.py [hunk 5]
```diff
@@ -58,7 +58,7 @@
 
 
 class EngineInstance:
-    def __init__(self, count: int, hosts: int = 50, srv_by_host: int = 20):
+    def __init__(self, count: int, hosts: int = 50, srv_by_host: int = 20,alias:bool = True):
```
Comment (bouda1) [inline tests/resources/Engine.py:61]:
```suggestion
    def __init__(self, count: int, hosts: int = 50, srv_by_host: int = 20, alias:bool = True):
```

File: tests/resources/Engine.py [hunk 6]
```diff
@@ -68,7 +68,7 @@ def __init__(self, count: int, hosts: int = 50, srv_by_host: int = 20):
         self.instances = count
         self.service_cmd = {}
         self.anomaly_detection_internal_id = 1
-        self.build_configs(hosts, srv_by_host)
+        self.build_configs(hosts, srv_by_host,alias=alias)
```
Comment (bouda1) [inline tests/resources/Engine.py:71]:
```suggestion
        self.build_configs(hosts, srv_by_host, alias=alias)
```

File: tests/resources/Engine.py [hunk 7]
```diff
@@ -186,7 +186,7 @@ def _create_centengine(self, id: int, debug_level=0):
                 "check_service_freshness=1\n"
                 "enable_flap_detection=0\n").format(id, debug_level, CONF_DIR, VAR_ROOT, ETC_ROOT, grpc_port)
 
-    def _create_host(self):
+    def _create_host(self,add_alias=True):
```
Comment (bouda1) [inline tests/resources/Engine.py:189]:
```suggestion
    def _create_host(self, add_alias=True):
```

File: tests/resources/Engine.py [hunk 8]
```diff
@@ -552,7 +556,7 @@ def create_tags(poller: int, nb: int, offset: int, tag_type: str):
 """
             ff.write(content)
 
-    def build_configs(self, hosts: int, services_by_host: int, debug_level=0):
+    def build_configs(self, hosts: int, services_by_host: int, debug_level=0,alias=True):
```
Comment (bouda1) [inline tests/resources/Engine.py:559]:
```suggestion
    def build_configs(self, hosts: int, services_by_host: int, debug_level=0, alias=True):
```

File: tests/resources/Engine.py [hunk 9]
```diff
@@ -517,14 +520,15 @@ def create_template_file(poller: int, typ: str, what: str, ids):
             content = ""
             idx = 1
             for i in ids:
-                content += """define {} {{
-name                   {}_template_{}
-{}               {}
+                content += f"""define {typ} {{
+name                   {typ}_template_{idx}
+alias                  alis_template{idx}
```
Comment (bouda1) [inline tests/resources/Engine.py:525]:
```suggestion
alias                  alias_template_{idx}
```

File: tests/resources/Engine.py [hunk 10]
```diff
@@ -517,14 +520,15 @@ def create_template_file(poller: int, typ: str, what: str, ids):
             content = ""
             idx = 1
             for i in ids:
-                content += """define {} {{
-name                   {}_template_{}
-{}               {}
+                content += f"""define {typ} {{
```
Comment (bouda1) [inline tests/resources/Engine.py:523]: A template should be always defined without alias. It is a bad practice to define it in a template.
So No need to modify this function here.

File: common/engine_conf/parser.cc (hunk 1 above)
Comment (sechkem) [reply common/engine_conf/parser.cc:711]: will fix

**Review by jean-christophe81 — APPROVED** [review]

File: tests/broker-engine/hosts-with-notes-and-actions.robot (hunk 2 above)
Comment (sechkem) [reply tests/broker-engine/hosts-with-notes-and-actions.robot:118]: will fix

File: common/engine_conf/host_helper.cc [hunk 11]
```diff
@@ -66,7 +66,15 @@ bool host_helper::hook(std::string_view key, const std::string_view& value) {
   /* Since we use key to get back the good key value, it is faster to give key
    * by copy to the method. We avoid one key allocation... */
   key = validate_key(key);
-  if (key == "contactgroups") {
+  if (key == "host_name"){
```
Comment (bouda1) [inline common/engine_conf/host_helper.cc:69]: coding style

File: common/engine_conf/host_helper.cc [hunk 12]
```diff
@@ -66,7 +66,15 @@ bool host_helper::hook(std::string_view key, const std::string_view& value) {
...
+    obj->set_host_name(std::string(value));
+    set_changed(obj->descriptor()->FindFieldByName("host_name")->index());
+    if(obj->alias().empty()){
+      obj->set_alias(std::string(value));
+      set_changed(obj->descriptor()->FindFieldByName("alias")->index());
+    }
+    return true;
+  }else if (key == "contactgroups") {
```
Comment (bouda1) [inline common/engine_conf/host_helper.cc:77]: coding style

File: common/engine_legacy_conf/host.cc [hunk 13]
```diff
@@ -1348,6 +1348,9 @@ bool host::_set_host_id(uint64_t value) {
  */
 bool host::_set_host_name(std::string const& value) {
   _host_name = value;
+  // if alias is empty we take the host name, better than taking template alias
+  if(_alias.empty())
```
Comment (bouda1) [inline common/engine_legacy_conf/host.cc:1352]: coding style

File: tests/broker-engine/hosts-with-notes-and-actions.robot [hunk 14]
```diff
@@ -105,3 +105,71 @@ EBSN3
     Should Be Equal As Strings    ${output}    (('${n}',),)
     Ctn Stop engine
     Ctn Kindly Stop Broker
+
+EBSN4
+    [Documentation]    New hosts with No Alias / Alias and have A Template
+    [Tags]    broker    engine    hosts
```
Comment (bouda1) [inline tests/broker-engine/hosts-with-notes-and-actions.robot:111]: Add the MON-.... in the tags

File: tests/resources/Engine.py [hunk 15]
```diff
@@ -930,6 +929,50 @@ def ctn_engine_config_set_value_in_hosts(idx: int, desc: str, key: str, value: s
     with open(filename, "w") as f:
         f.writelines(lines)
 
+def ctn_engine_config_delete_value_in_hosts(idx: int, desc: str, key: str, file: str = 'hosts.cfg'):
+    """
+    Set a parameter in the hosts.cfg for the Engine configuration idx.
```
Comment (bouda1) [inline tests/resources/Engine.py:934]: Wrong comment

File: common/engine_conf/host_helper.cc (hunk 11 above)
Comment (sechkem) [reply common/engine_conf/host_helper.cc]: will fix

File: common/engine_conf/host_helper.cc (hunk 12 above)
Comment (sechkem) [reply common/engine_conf/host_helper.cc]: will fix

File: tests/resources/Engine.py (hunk 15 above)
Comment (sechkem) [reply tests/resources/Engine.py]: will fix

File: tests/broker-engine/hosts-with-notes-and-actions.robot (hunk 14 above)
Comment (sechkem) [reply tests/broker-engine/hosts-with-notes-and-actions.robot]: will fix

**Review by bouda1 — APPROVED** [review]

**Pull request merged**